src/seat.c

```diff
@@ -506,13 +506,51 @@ seat_finish(struct server *server)
 }
 
 static void
-configure_keyboard(struct wlr_input_device *device)
+reset_window_keyboard_layout_groups(struct server *server)
 {
+	if (!rc.kb_layout_per_window) {
+		return;
+	}
+
+	/*
+	 * We don't want to reset all window keyboard-layout groups on every
+	 * reconfigure unless XKB_DEFAULT_LAYOUT has actully changed.
+	 */
+	static char last_layout[64];
+	char *current_layout = getenv("XKB_DEFAULT_LAYOUT") ? : "";
```

Comment:
Recommend avoiding the GNU extension to `?:`.

Reply:
Done